File: snobot_sim/cpp/lib/SnobotSim/MotorSim/StaticLoadDcMotorSim.cpp
/*
 * StaticLoadDcMotorSim.cpp
 *
 *  Created on: May 9, 2017
 *      Author: preiniger
 */

#include "SnobotSim/MotorSim/StaticLoadDcMotorSim.h"

StaticLoadDcMotorSim::StaticLoadDcMotorSim(const DcMotorModel& aMotorModel, double aLoad, double aConversionFactor) :
        BaseDcMotorSimulator("Static Load", aMotorModel, aConversionFactor), mLoad(aLoad)
{

}

StaticLoadDcMotorSim::~StaticLoadDcMotorSim()
{

}

void StaticLoadDcMotorSim::Update(double cycleTime)
{
    mMotorModel.Step(mVoltagePercentage * 12, mLoad, 0, cycleTime);
}

File: snobot_sim_java/java/snobot_sim_java/com/snobot/simulator/motor_sim/StaticLoadDcMotorSim.java
package com.snobot.simulator.motor_sim;

public class StaticLoadDcMotorSim extends BaseDcMotorSimulator
{
    protected final StaticLoadMotorSimulationConfig mConfig;

    public StaticLoadDcMotorSim(DcMotorModel aModel, StaticLoadMotorSimulationConfig aConfig)
    {
        super(aModel, aConfig.mConversionFactor);

        mConfig = aConfig;
    }

    @Override
    public void update(double cycleTime)
    {
        mMotorModel.step(mVoltagePercentage * 12, mConfig.mLoad, 0, cycleTime);
    }

    public StaticLoadMotorSimulationConfig getConfig()
    {
        return mConfig;
    }

}
